src/audiomixerboard.cpp

```diff
@@ -1264,6 +1268,52 @@ void CAudioMixerBoard::SetAllFaderLevelsToNewClientLevel()
     }
 }
 
+void CAudioMixerBoard::SetAllPanLeftRight() 
+{
+    QMutexLocker locker ( &Mutex );
+
+    int iNumConnectedClients = veciChannelOrder.size();
+
+    if ( iNumConnectedClients < 3) {
+        return;
+    }
+    
+    int iCount = 0;
+    int halfCount = iNumConnectedClients / 2;
+
+    for ( const int iChannelID : veciChannelOrder )
+    {
+        // set my channel to middle
+        if ( iChannelID == iMyChannelID )
+        {
+            vecpChanFader[iChannelID]->SetPanValue ( 50 );
+            continue;
+        }
+
+        // leave a spot in the middle for My Channel
+        if ( iCount == halfCount ) 
+        {
+            iCount += 2;
+        }
+
+        vecpChanFader[iChannelID]->SetPanValue ( iCount++ * AUD_MIX_PAN_MAX / iNumConnectedClients );
+    }
+}
+
+void CAudioMixerBoard::SetAllPanRandom()
+{
+    QMutexLocker locker ( &Mutex );
```

Comment:
Not sure about that (not a Qt expert), but do you need to unlock it?

Reply:
The mutexlocker unlocks the mutex automatically when it goes out of scope (RAII)